### ISSUE TITLE
Unable to read <null> decimal with SQLite

### ISSUE THREAD

**Issue by marceloverdijk** [prisma/prisma#12497]
### Bug description

Having the following field in a SQLite database:

    scheduled_distance NUMERIC (6, 3)

and in schema.prisma:

    scheduledDistance Decimal?   @map("scheduled_distance")

will give:

```
Invalid `prisma.race.findMany()` invocation:\n\n\n  
Inconsistent column data: Conversion failed: Value NUMERIC(6,3) not supported
```

in case a record contains a `null` value for this column.

So e.g. value `806.982`, `806` works, but `null` gives the above error. Note I mean actual `null` not a `"null"` string.

As it is an optional column `null` should be possible.

### How to reproduce

See bug description.

### Expected behavior

Rows with e.g. `null` to be read successfully.

### Prisma information

See bug description.

### Environment & setup

- OS: Mac OS
- Database: SQLite
- Node.js version: v17.6.0

### Prisma Version

```
3.11.0
```

**Comment by marceloverdijk** [comment]
[prisma-decimal-null-sqlite.zip](https://github.com/prisma/prisma/files/8339759/prisma-decimal-null-sqlite.zip)

Attached an example that reproduces the issue.
It is based on the Prisma starter, where User model has been extended with a `latitude` and `longitude` `Decimal?` field in the Prisma schema, and a NUMERIC(10, 6) column in the SQLite database.

**Comment by janpio** [comment]
SQLite's "type system" is always ready for a surprise.

Just to double check: You had the SQLite database already and wrote a Prisma schema manually to get to this mismatch, correct?

**Comment by marceloverdijk** [comment]
Yes correct, existing SQLite database with Prisma introspection.

**Comment by janpio** [comment]
Ooooookay, if this was created via Introspection then we have a problem.

**Comment by marceloverdijk** [comment]
Actually I don’t think there is a Prisma problem.. the introspection resulting in `Decimal?` is correct I think. The issue was in the SQLite database which was not created by Prisma.

**Comment by janpio** [comment]
But if the database is `scheduled_distance NUMERIC (6, 3)`, which is not actually valid for SQL from what you wrote, why should it introspect this as `Decimal?`? Wouldn't it be better to not introspect that field then?

**Comment by marceloverdijk** [comment]
@janpio good point indeed... I introspected with older version some time ago. So probably good to see if latest version is doing that as well.
